src/dmd/root/array.h

```diff
@@ -240,4 +240,17 @@ struct BitArray
     BitArray(const BitArray&);
 };
 
+// This is the type that the D compiler will mangle D slices as when mangling
+// for C++. This type is ABI compatible with native D slices.
```

Comment:
Is ABI compatibility actually tested somewhere? I seriously doubt it is on Win64 for example. LDC on Win64 returns and passes slices as pairs (in 2 registers if available), whereas a 128-bit struct is passed indirectly by value (hidden ref), and there is no way for a struct to emulate that behaviour.

[Btw, the `__dslice` ctors below make MSVC++ treat this as a non-POD and return it via hidden pointer, but that's irrelevant here, as it's > 64 bits anyway.]

Reply:
> Is ABI compatibility actually tested somewhere?

No.

Reply:
> LDC on Win64 returns and passes slices (and delegates IIRC) as pairs (in 2 registers if available), whereas a 128-bit struct is passed/returned indirectly by value (hidden ref), and there is no way for a struct to emulate that behavior.

That's unfortunate.